File: frontend/src/shared/config/imageUpload.ts
/**
 * 画像アップロードで受け入れる MIME タイプの単一ソース。
 *
 * プロフィール画像・AI チャット・リッチテキストエディタなどで共通に使う。
 * `<input accept>` と JS 側の絞り込みで同じ定義を参照し、ずれ（バックエンド非互換 MIME の受理）を防ぐ。
 */
export const ACCEPTED_IMAGE_MIME_TYPES = [
  'image/png',
  'image/jpeg',
  'image/gif',
  'image/webp',
] as const;

/** `<input type="file" accept>` に渡す文字列（カンマ区切り）。 */
export const ACCEPTED_IMAGE_ACCEPT_ATTR = ACCEPTED_IMAGE_MIME_TYPES.join(',');

/** file.type が受け入れ対象の画像 MIME かを返す。 */
export function isAcceptedImageMimeType(type: string): boolean {
  return (ACCEPTED_IMAGE_MIME_TYPES as readonly string[]).includes(type);
}

/**
 * 画像アップロードの上限バイト数（10 MiB）。backend の上限と一致させる。
 *
 * サーバー側の判定を待たず、選んだ直後にクライアント側でも早期に弾く（無駄な
 * アップロード開始と、待たされた末の失敗表示を避けるため）。
 */
export const MAX_IMAGE_UPLOAD_BYTES = 10 * 1024 * 1024;

File: backend/internal/domain/image_upload.go
package domain

import "errors"

// AcceptedImageContentTypes は画像アップロードで受け付ける Content-Type の許可リスト。
// ここに無い型は ValidateImageUpload が ErrUnsupportedImageContentType で弾く
// （FRESTYLE-9: 以前は contentType を検証せずそのまま S3 の PUT presign に焼き込んでいた）。
var AcceptedImageContentTypes = map[string]bool{
	"image/png":  true,
	"image/jpeg": true,
	"image/gif":  true,
	"image/webp": true,
}

// MaxImageUploadBytes は画像アップロード 1 件あたりのサイズ上限（10 MiB）。
const MaxImageUploadBytes = 10 * 1024 * 1024

// ErrUnsupportedImageContentType は許可リストに無い Content-Type が指定されたときに返す。
var ErrUnsupportedImageContentType = errors.New("unsupported image content type")

// ErrImageTooLarge は size が 0 以下、または上限を超えるときに返す。
var ErrImageTooLarge = errors.New("image is too large")

// ValidateImageUpload は画像アップロード要求（Content-Type とサイズ）が保存してよい形かを見る。
// presign の前に必ず通す入口で、不正なら presign 自体を行わない
// （不正な値のまま署名を発行すると、上限の無い PUT を許すのと同じ穴になる）。
func ValidateImageUpload(contentType string, size int64) error {
	if !AcceptedImageContentTypes[contentType] {
		return ErrUnsupportedImageContentType
	}
	if size <= 0 || size > MaxImageUploadBytes {
		return ErrImageTooLarge
	}
	return nil
}
